format-delimited/src/main/java/io/cdap/plugin/format/delimited/input/PathTrackingDelimitedInputFormat.java

```diff
@@ -80,10 +80,14 @@ public NullWritable getCurrentKey() {
 
         StructuredRecord.Builder builder = StructuredRecord.builder(schema);
         Iterator<Schema.Field> fields = schema.getFields().iterator();
+        Iterable<String> splitDelimitedString = Splitter.on(delimiter).split(delimitedString);
+        int numDataFields = 0;
+        for (String part : splitDelimitedString) {
```

Comment:
This will cause an extra loop through every record. It's better to only do this in the error scenario. 

That being said, I don't understand when the Splitter would return a different number of elements than split(). What would be an example of when they are different?

Reply:
When splitting this string "2018,9,111,NIXON,,JESSE,02/01/1840,Database Engineer,Baseball Operations,New York Yankees,07/07/2014,,,,Male,,,White,Baseball Ops,,\"Group 3 - Professionals, Technicians, Supervisors\",Active,,"; the output from Splitter is 26 fields but the output from split() is 24 fields which I think is because the latter automatically ignored the last delimitter.

Reply:
Added the numDataFields loop in the error scenario.